lib/registry.js

```diff
@@ -16,7 +17,9 @@ export function define(id, spec) {
   if (get(id)) {
     console.warn(`Overwriting spec ${getName(id)}`)
   }
-  spec.name = getName(id)
+  if (!spec.name && typeof spec !== "symbol") {
```

Comment:
> I'll check the predicates; perhaps use lodash?

Currently they're backed by lodash anyways. I haven't been very consistent myself in this regard, but it would be awesome if typical checks (like not null, types etc) would use the predicates we also expose. Just as an eat-your-own-dogfood style of working.

> Symbol is still pretty new to ES.

Except for Android <=5.0 it's [supported](http://kangax.github.io/compat-table/es6/#test-Symbol) in all major JS runtimes, I don't see a reason not to recommend it.

> Another thing :)

Ha, good catch indeed :grin: I agree with your proposal, it should check the registry directly.